WordPress/Classes/ViewRelated/Gutenberg/Layout Picker/GutenbergLayoutPickerViewController.swift

```diff
@@ -0,0 +1,193 @@
+import UIKit
+import Gridicons
+
+class GutenbergLayoutPickerViewController: UIViewController {
+
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var closeButton: UIButton!
+    @IBOutlet weak var titleView: UILabel!
+    @IBOutlet weak var largeTitleView: UILabel!
+    @IBOutlet weak var promptView: UILabel!
+    @IBOutlet weak var categoryBar: UICollectionView!
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var footerView: UIView!
+    @IBOutlet weak var createBlankPageBtn: UIButton!
+    @IBOutlet weak var headerHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var titleToSubtitleSpacing: NSLayoutConstraint!
+    @IBOutlet weak var subtitleToCategoryBarSpacing: NSLayoutConstraint!
+    @IBOutlet weak var minHeaderBottomSpacing: NSLayoutConstraint!
+    @IBOutlet weak var maxHeaderBottomSpacing: NSLayoutConstraint!
+
+    var completion: PageCoordinator.TemplateSelectionCompletion? = nil
+
+    private var maxHeaderHeight: CGFloat {
+        return largeTitleView.frame.height +
+        midHeaderHeight
+    }
+    private var midHeaderHeight: CGFloat {
+        return titleToSubtitleSpacing.constant +
+            promptView.frame.height +
+            subtitleToCategoryBarSpacing.constant +
+            categoryBar.frame.height +
+            maxHeaderBottomSpacing.constant
+    }
+    private var minHeaderHeight: CGFloat {
+        return categoryBar.frame.height + minHeaderBottomSpacing.constant
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        closeButton.setImage(UIImage.gridicon(.crossSmall), for: .normal)
+        styleButtons()
+        layoutHeader()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        (navigationController as? GutenbergLightNavigationController)?.shadowIsHidden = true
+        super.viewWillAppear(animated)
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        (navigationController as? GutenbergLightNavigationController)?.shadowIsHidden = false
+        super.viewDidDisappear(animated)
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        (navigationController as? GutenbergLightNavigationController)?.shadowIsHidden = false
+        super.prepare(for: segue, sender: sender)
+    }
```

Comment:
There aren't any segues on the screen yet but they will be added with a preview functionality later. I wanted to add this functionality now so it wouldn't be forgotten when that work is done.